### PR TITLE
Tune Tune Sentiment - Inference Upload

### PR DESCRIPTION
**Pull request by yidinghabana**
One new folder uploaded for Fine Tune Sentiment Inference, since the Try it Live issue got solved:

* ft-fine-tuning-inference/

Two Documents uploaded:

*        README_inference.md
*        sentiment_analysis_inference.yaml

### CODE REVIEW

File: ft-fine-tuning-inference/inference.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    sentiment_label = ['negative', 'neutral', 'positive']
+
+    input_data = data["text"]
+    data = pd.DataFrame([input_data], columns=["text"])
+    data_input=data["text"]
+
+    data_input = data_input.apply(lambda x: preprocess(x))
+
+    # os.chdir("fine-tune-inference/ft-fine-tuning-inference")
+    os.environ['CUDA_VISIBLE_DEVICES'] = "0"
```
Comment (orionsBeltWest) [inline ft-fine-tuning-inference/inference.py]: recommend having the hard coded values in lines: 58, 61, and 63 passed in the predict function argument with a default value as used here.

File: ft-fine-tuning-inference/inference.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        # input_ids = torch.tensor(self.encodings['input_ids'])
+        # target_ids = torch.tensor(self.labels[idx])
+        # return {"input_ids": input_ids, "labels": target_ids}
+
+    def __len__(self):
+        return len(self.encodings["input_ids"])
+
+
+# Define predict function
+def predict(data):
```
Comment (orionsBeltWest) [inline ft-fine-tuning-inference/inference.py:86]: who call predict(data)?

File: ft-fine-tuning-inference/inference.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+# print('====json_dump====', json_dump)
+
+# data = json.loads(json_dump)
+# print(type(data["text"]))
+
+# if __name__ == "__main__":
+#     result = predict(data)
+#     print(result)
+
+
```
Comment (orionsBeltWest) [inline ft-fine-tuning-inference/inference.py:149]: I do not see a unit test cases?

**Review by orionsBeltWest — CHANGES_REQUESTED** [review]
Code structured okay however, could you check the items highlighted in my comments

File: ft-fine-tuning-inference/inference.py (hunk 1 above)
Comment (yidinghabana) [reply ft-fine-tuning-inference/inference.py]: I uploaded a new version for review by changing accordingly.

File: ft-fine-tuning-inference/inference.py (hunk 2 above)
Comment (yidinghabana) [reply ft-fine-tuning-inference/inference.py:86]: The webservice type of library yaml will call main function predict()

image

File: ft-fine-tuning-inference/inference.py (hunk 3 above)
Comment (yidinghabana) [reply ft-fine-tuning-inference/inference.py:149]: I copied my unittests file and uploaded for your PR review as well.

File: ft-fine-tuning-inference/inference.py (hunk 1 above)
Comment (orionsBeltWest) [reply ft-fine-tuning-inference/inference.py]: I do not see the changes. Let's walk this together ?

File: ft-fine-tuning-inference/inference.py (hunk 3 above)
Comment (orionsBeltWest) [reply ft-fine-tuning-inference/inference.py:149]: okay I see the changes.

Comment (orionsBeltWest) [reply ft-fine-tuning-inference/inference.py:149]: looks okay

**Review by orionsBeltWest — APPROVED** [review]

**Comment by orionsBeltWest** [comment]
Looks good

**Review by Ramji-Satya — APPROVED** [review]

**Pull request merged**